src/main/java/carpet/CarpetSettings.java

```diff
@@ -682,6 +682,47 @@ public String description() {
     )
     public static int viewDistance = 0;
 
+    private static class SimulationDistanceValidator extends Validator<Integer>
+    {
+        @Override public Integer validate(ServerCommandSource source, ParsedRule<Integer> currentRule, Integer newValue, String string)
+        {
+            if (currentRule.get().equals(newValue) || source == null)
+            {
+                return newValue;
+            }
+            if (newValue < 0 || newValue > 32)
+            {
+                Messenger.m(source, "r simulation distance has to be between 0 and 32");
+                return null;
+            }
+            MinecraftServer server = source.getServer();
+
+            if (server.isDedicated())
+            {
+                int vd = (newValue >= 2)?newValue:((DedicatedServer) server).getProperties().simulationDistance;
```

Comment:
Is the minimum simulation distance 2? This minimum could also probably be moved above so setting it to values in the middle doesn't silently ignore the user's choice.

Reply:
I just copied and pasted the implementation for viewDistance and that should also happen for viewDistance as well.